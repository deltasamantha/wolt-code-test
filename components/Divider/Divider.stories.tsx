import {Meta, StoryObj} from "@storybook/react-native";
import {View} from "react-native";

import Divider, {Props} from "./Divider";
import {storyStyles} from "./Divider.styles";

const DividerMeta: Meta<Props> = {
  title: "Divider",
  component: Divider,
  argTypes: {
    type: {
      description: "Divider color",
      control: "radio",
      options: ["light", "dark"],
    },
  },
  decorators: [
    Story => (
      <View style={storyStyles.rootContainer}>
        <Story />
      </View>
    ),
  ],
};

export default DividerMeta;

export const Primary: StoryObj<Props> = {
  args: {
    type: "dark",
  },
};

export const Light: StoryObj<Props> = {
  args: {
    type: "light",
  },
};
