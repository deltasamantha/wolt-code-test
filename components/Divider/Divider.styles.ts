import {StyleSheet} from "react-native";

const DIVIDER_HEIGHT = 1;

export const styles = StyleSheet.create({
  divider: {
    height: DIVIDER_HEIGHT,
    width: "100%",
  },
});
