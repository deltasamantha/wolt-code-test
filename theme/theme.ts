const theme = {
  colors: {
    colorBackground: "#ABD6EA",
    colorPrimary: "",
    colorAccent: "#5BCB02",
    colorBlack: "#000000",
    colorWhite: "#FFFFFF",
    fontColorDark: "#202125",
    fontColorLight: "#A1A2A4",
  },
  borderRadius: {
    sm: 5,
    md: 10,
    lg: 20,
  },
  spacing: {
    xs: 2,
    sm: 4,
    md: 10,
    lg: 16,
    xl: 24,
  },
  fontSize: {
    small: 12,
    medium: 16,
    large: 24,
  },
  fontLineHeight: {
    small: 16,
    medium: 22,
    large: 30,
  },
};

export default theme;
