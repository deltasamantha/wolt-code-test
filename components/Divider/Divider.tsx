import React from "react";
import {View} from "react-native";

import {styles} from "./Divider.styles";
import {getDividerColor} from "../../utils/DividerUtils";

export type DividerType = "light" | "dark";

/**
 * Prop definition for Divider component
 */
export interface Props {
  /** Color type for the divider, could be one of `"light" | "dark"` */
  type?: DividerType;
}

/**
 * ## Divider component
 * -------------
 * @description Component to render a horizontal divider line.
 */
const Divider: React.FC<Props> = ({type = "dark"}: Props) => {
  return (
    <View
      testID="divider"
      style={[styles.divider, {backgroundColor: getDividerColor(type)}]}
    />
  );
};

export default Divider;
