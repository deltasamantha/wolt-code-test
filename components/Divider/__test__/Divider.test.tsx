import {render, cleanup, screen} from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";

import theme from "../../../theme/theme";
import Divider from "../Divider";

afterEach(cleanup);

describe("Divider", () => {
  it("Should render Divider properly", () => {
    const {toJSON} = render(<Divider />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Should render dark variant", () => {
    const color = theme.colors.colorDark;
    render(<Divider type="dark" />);
    const divider = screen.getByTestId("divider");
    expect(divider).toBeTruthy();
    expect(divider).toHaveStyle({
      backgroundColor: color,
    });
  });

  it("Should render light variant", () => {
    const color = theme.colors.colorExtraLight;
    render(<Divider type="light" />);
    const divider = screen.getByTestId("divider");
    expect(divider).toBeTruthy();
    expect(divider).toHaveStyle({
      backgroundColor: color,
    });
  });
});
