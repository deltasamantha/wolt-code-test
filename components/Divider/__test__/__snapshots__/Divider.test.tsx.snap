// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Should render Divider properly 1`] = `
<View
  style={
    [
      {
        "height": 1,
        "width": "100%",
      },
      {
        "backgroundColor": "#202125",
      },
    ]
  }
  testID="divider"
/>
`;
